### DISCUSSION TITLE
Volume Exponential Moving Average

### DISCUSSION THREAD

**Discussion by myalgomate** [DaveSkender/Stock.Indicators#568] [Ideas]
Hi,
Thanks for your valuable library.
I would like to share a thought if we can make a generic indicator like you already have an EMA indicator for EMA Results.
you are using a candle close to calculate the same.
Now in case, I need EMA on volume then I will need to change the src of the EMA indicator.
Instead, can we have an argument to use the input as close/volume likewise

Or if it's not possible can you please add Volume Exponential Moving Average

**Comment by DaveSkender** [comment]
As long as your historical quotes contain volume information, EMA on Volume can be done like this:

```csharp
quotes.getEma(14, CandlePart.Volume);
```

Incidentally, since we augmented SMA to accept `CandlePart` these two are also equivalent:

```csharp
quotes.getSma(14, CandlePart.Volume);
quotes.getVolSma(14);
```

These `CandlePart` modifiers were added in v1.7.0

**Reply by myalgomate** [reply]
Thanks.
